src/wp-admin/includes/class-wp-automatic-updater.php

```diff
@@ -158,6 +158,12 @@ public function should_update( $type, $item, $context ) {
 		// Next up, is this an item we can update?
 		if ( 'core' === $type ) {
 			$update = Core_Upgrader::should_update_to_version( $item->current );
+		} elseif ( 'plugin' === $type || 'theme' === $type ) {
+			$update = false;
+			if ( wp_is_auto_update_enabled_for_type( $type ) ) {
+				$auto_updates = (array) get_site_option( "auto_update_{$type}s", array() );
+				$update       = in_array( $item->{$type}, $auto_updates, true ) || ! empty( $item->autoupdate );
+			}
 		} else {
 			$update = ! empty( $item->autoupdate );
 		}
```

Comment:
May be reading it wrong but seems this will still prevent pushing security updates from the API? The core settings are the "default", plugins and the API should be able to override them. Seems that should be something like:
```
if ( 'core' === $type ) {
	$update = Core_Upgrader::should_update_to_version( $item->current );
} else {
	$update = ! empty( $item->autoupdate );

	if (  ! $update && ( 'plugin' === $type || 'theme' === $type ) ) {
		// Use the default settings as set by the user
	}
}
```

Reply:
yes, see the comment from Timothy on the old PR, https://github.com/WordPress/wordpress-develop/pull/279#discussion_r426836326

Reply:
Ah, missed that, sorry :)